src/KurrentDB.Api.V2/Modules/CustomIndexes/Validators/NameValidator.cs

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) Kurrent, Inc and/or licensed to Kurrent, Inc under one or more agreements.
+// Kurrent, Inc licenses this file to you under the Kurrent License v1 (see LICENSE.md).
+
+using FluentValidation;
+using KurrentDB.Api.Infrastructure.FluentValidation;
+
+namespace KurrentDB.Api.Modules.CustomIndexes.Validators;
+
+// Lower case restriction because identifiers are case-insensitive in duckdb
+class NameValidator : ValidatorBase<NameValidator, string?> {
+	public NameValidator() =>
+		RuleFor(x => x)
+			.MinimumLength(1)
+			.MaximumLength(1000)
```

Comment:
My take on these   limits:

* this is an identifier not a description so I would say that max length of 1000 is way too much.
* minimum should be at least 2 chars.

Reply:
for later then. easy improvement if agreed